src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -931,7 +931,21 @@ void c_typecheck_baset::typecheck_expr_sizeof(exprt &expr)
   }
   else
   {
-    type.swap(to_unary_expr(expr).op().type());
+    const exprt &op = to_unary_expr(as_const(expr)).op();
+    // This is one of the few places where it's detectable
+    // that we are using "bool" for boolean operators instead
+    // of "int". We convert for this reason.
+    if(
+      op.type().id() == ID_bool &&
+      (op.id() == ID_not || op.id() == ID_and || op.id() == ID_or ||
+       op.id() == ID_equal || op.id() == ID_notequal || op.id() == ID_lt ||
+       op.id() == ID_le || op.id() == ID_gt || op.id() == ID_ge ||
+       op.id() == ID_if))
```

Comment:
I would probably drop the check of the operator.
Say ?: is missing.

Reply:
There's `ID_if` in the above, but I agree this could easily be missing some other operator. When writing this code I was wondering whether to just always cast, or whether some cases should be considered illformed input and be rejected by the type checker?

Reply:
I think it's the right thing to always use `int` whenever "proper bool" is encountered in `sizeof`.

Reply:
Ok, done!